### PR TITLE
Check csproj files for adjacent packages.config.

### PR DESCRIPTION
**Pull request by sebasgomez238**
Although the nuget updater was properly running the code to update packages.config files, the `file_fetcher.rb` was only reporting the packages.config in the directory where the job started. 

This PR adds logic to:
1. Look for `packages.config` files adjacent to imported project files.
2. Modifies the `fetch_imported_property_files` method to improve fetching imported files.

The reason for the second change is that when for example `projectA.csproj` has references to both `projectB.csproj` and `projectC.csproj` then the imported project files was incorrectly reporting ONLY `projectC.csproj`.

All feedback is welcome.

### CODE REVIEW

**Comment by brettfo** [comment]
Add the following to `nuget/.gitignore` and you can remove the `bin` and `obj` directories that were added.

``` diff
 /.bundle/
 !.bundle/config
 /.env
 /tmp
 /dependabot-*.gem
 Properties/
+bin/
+obj/
```

File: nuget/spec/fixtures/github/csproj_in_subdirectory/src/some-project/obj/Debug/net8.0/some-project.GeneratedMSBuildEditorConfig.editorconfig
Comment (JamieMagee) [inline nuget/spec/fixtures/github/csproj_in_subdirectory/src/some-project/obj/Debug/net8.0/some-project.GeneratedMSBuildEditorConfig.editorconfig:1]: Is this file meant to be included?

Comment (sebasgomez238) [reply nuget/spec/fixtures/github/csproj_in_subdirectory/src/some-project/obj/Debug/net8.0/some-project.GeneratedMSBuildEditorConfig.editorconfig:1]: Thanks! I thought I had deleted all of them. I don't know why these got generated.

File: nuget/lib/dependabot/nuget/file_fetcher.rb [hunk 1]
```diff
@@ -363,23 +386,36 @@ def fetch_imported_property_files(file:, previously_fetched_files:)
...
+                file: fetched_file,
+                previously_fetched_files: previously_fetched_files + [file]
+              )
+
+              # Add fetched file and grandchild property files to the set
+              fetched_files_set << fetched_file
+              fetched_files_set.merge(grandchild_property_files)
+            rescue Dependabot::DependencyFileNotFound
+              # Don't worry about missing files, just skip them for now
+              nil
```
[the remark this answers is not in the document]
Comment (sebasgomez238) [reply nuget/lib/dependabot/nuget/file_fetcher.rb]: Done!

**Review by brettfo — APPROVED** [review]

**Review by brettfo — APPROVED** [review]

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**